src/material/slide-toggle/slide-toggle-required-validator.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  Directive,
+  forwardRef,
+  Provider,
+} from '@angular/core';
+import {
+  CheckboxRequiredValidator,
+  NG_VALIDATORS,
+} from '@angular/forms';
+
+export const MAT_SLIDE_TOGGLE_REQUIRED_VALIDATOR: Provider = {
+  provide: NG_VALIDATORS,
+  useExisting: forwardRef(() => MatSlideToggleRequiredValidator),
+  multi: true
+};
+
+/**
+ * Validator for Material slide-toggle components with the required attribute in a
+ * template-driven form. The default validator for required form controls asserts
+ * that the control value is not undefined but that is not appropriate for a slide-toggle
+ * where the value is always defined.
+ *
+ * Required slide-toggle form controls are valid when checked.
+ */
+@Directive({
+  selector: `mat-slide-toggle[required][formControlName],
+             mat-slide-toggle[required][formControl], mat-slide-toggle[required][ngModel]`,
+  providers: [MAT_SLIDE_TOGGLE_REQUIRED_VALIDATOR],
+  host: {'[attr.required]': 'required ? "" : null'}
```

Comment:
Isn't this attribute supposed to be on the `input[type="checkbox"]`?

Reply:
Yeah. I'm not sure why this is needed. I took that one from the `mat-checkbox` required validator. I'd actually think it makes more sense the other way around where we remove it from the host element.

Reply:
@crisbeto I removed it for now. I don't see why it needs to be like that. It might be worth to consider removing it for the `MatCheckboxRequiredValidator` as well.

Reply:
I'm guessing the `MatCheckbox` one was copied from the one in core. The one in core is supposed to be put directly on the input.

Reply:
Yeah that would explain why it has that host binding. I've created a PR to remove it for the checkbox.